test/tests/ics/integral_preserving_function_ic/tests

```diff
@@ -0,0 +1,24 @@
+[Tests]
+  issues = '#19476'
+  design = 'IntegralPreservingFunctionIC.md'
+  [missing_initial]
+    type = RunException
+    input = 'sinusoidal_z.i'
+    expect_err = "The 'execute_on' parameter for the 'vol' postprocessor must include 'initial'!"
+    cli_args = 'Postprocessors/vol/execute_on=final'
+    requirement = 'The postprocessor must have execute_on initial'
+  []
+  [zero_integral]
+    type = RunException
+    input = 'sinusoidal_z.i'
+    expect_err = "The integral of 'vol' cannot be zero!"
+    cli_args = 'Postprocessors/vol/function=0.0'
+    requirement = 'The postprocessor for normalization cannot be zero'
+  []
```

Comment:
rework these tests:
[errors]
  requirement = 'the system shall error if'
  [missing_initial]
  ...
   detail = 'the postprocessor used for normalization does not execute on initial, before setting the initial conditions'
   []
  [zero_integral]
  ....
  []
[]